### PR TITLE
Bug fixing on state/modes, combined fragments, seq messages

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
Change-Id: I0000000000000000000000000000000000000000
Signed-off-by: Georgiana Ecobici <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

**Review by claudiasima-ro — APPROVED** [review]

File: plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/DiagramToXtextCommands.java [hunk 1]
```diff
@@ -329,7 +352,7 @@ private static void generateDeactivatioOnMessages(ExecutionEnd executionEnd,
   }
 
   private static int generateMessages(Object[] ends, int i,
-      Stack<org.polarsys.capella.scenario.editor.dsl.textualScenario.SequenceMessage> messagesToDeactivate,
+      Stack<Message> messagesToDeactivate,
```
Comment (minhtutonthat) [inline plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/DiagramToXtextCommands.java]: Could you use fully qualified name for classes coming form the DSL to distinguish from those coming from Capella?

File: plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java [hunk 2]
```diff
@@ -200,27 +225,66 @@ private static boolean reorderParticipants(List<InstanceRole> instanceRoles, ELi
...
+            && timelapse.getStart().getCoveredInstanceRoles().get(0).getName().equals(participantName))
+        .collect(Collectors.toList());
+    
+    for (TimeLapse timeLapse : stateFragmentsToBeDeleted) {
+      removeStateFragmentFromScenario(scenario, timeLapse);
+    } 
+  }
+
+  /**
+   * Remove from diagram all combined fragment related to the participant that has just been deleted
```
Comment (minhtutonthat) [inline plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java:285]: Please update the documentation of this method.

File: plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java [hunk 3]
```diff
@@ -346,102 +419,285 @@ private void reorderCapellaDeactivationMessages(Scenario scenario,
...
+      scenario.getOwnedInteractionFragments().add(interactionStateEnd);
+
+      StateFragment stateFragment = createStateFragment(interactionStateStart, interactionStateEnd, relatedElement,
+          xtextElement.getName());
+
+      ownedTimeLapses.add(stateFragment);
+    }
+  }
+
+  private static List<TimeLapse> getStateFragmentsWithGivenInstanceRoleAndRelatedElement(Scenario scenario,
```
Comment (minhtutonthat) [inline plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java]: Could you get a shorter name for the method and document what it does in the comment?

File: plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java [hunk 4]
```diff
@@ -346,102 +419,285 @@ private void reorderCapellaDeactivationMessages(Scenario scenario,
...
+  }
+
+  private static void editStateFragment(Scenario scenario,
+      org.polarsys.capella.scenario.editor.dsl.textualScenario.StateFragment xtextElement,
+      List<EObject> previousEditorStateFragments) {
+
+    InstanceRole instanceRole = EmbeddedEditorInstanceHelper.getInstanceRole(xtextElement.getTimeline());
+    EList<TimeLapse> ownedTimeLapses = scenario.getOwnedTimeLapses();
+
+    List<EObject> simillarStateFragments = previousEditorStateFragments.stream()
```
Comment (minhtutonthat) [inline plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java:712]: This is an example of the complexity of the code when processing edge cases without any comment. I just have no idea why do we have to lood at similar StateFragments while creating new StateFragment here.

**Review by minhtutonthat — CHANGES_REQUESTED** [review]
- Could you resolve Sonar issues in all the (not generated) code, especially:
	+ unnecessary if clauses
	+ unused parameters
	
- Could you go through method parameters' name (especially in the 2 classes DiagramToXtextCommands and XtextToDiagramCommands) and document them in the code because it's quite hard sometimes to follow and thus difficult to maintain later. This is just an example but could you do the same thing for other methods to make the code more readable? I would change 
org.polarsys.capella.scenario.editor.embeddededitor.commands.DiagramToXtextCommands.generateMessages(Object[] ends, 
      int i,
      Stack<Message> messagesToDeactivate,
      Stack<CombinedFragment> combinedFragments, Stack<Block> blockOperands, EList<EObject> elements,
      TextualScenarioFactory factory) 
to something like:
  /**
   * 
   * @param allCapellaEnds all input ends coming from the input diagram
   * @param messageEndPosition current message end position in all input ends
   * @param messagesToDeactivate to keep track of messages to deactivate
   * @param xtextBlockOperands to keep track of currently being processed output block operand
   * @param xtextElements the list of all output language elements
   * @param factory
   * @return the next input end to be processed
   */
org.polarsys.capella.scenario.editor.embeddededitor.commands.DiagramToXtextCommands.generateMessages(Object[] allCapellaEnds, 
      int messageEndPosition,
      Stack<Message> messagesToDeactivate,
      Stack<Block> xtextBlockOperands, 
	  EList<EObject> xtextElements,
      TextualScenarioFactory factory)
	
- Change the name of the method org.polarsys.capella.scenario.editor.embeddededitor.commands.DiagramToXtextCommands.generateActors(Model, Scenario, TextualScenarioFactory)
=> generateParticipants

- Similarly a lot of methods do not have comments about what they do even though their name do not give a clear idea either.

**Review by minhtutonthat — APPROVED** [review]

**Pull request merged**